pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java

```diff
@@ -411,6 +412,7 @@ public void copyToLocalFile(URI srcUri, File dstFile)
       throws Exception {
     LOGGER.info("Copy {} to local {}", srcUri, dstFile.getAbsolutePath());
     URI base = getBase(srcUri);
+    FileUtils.forceMkdir(dstFile.getParentFile());
```

Comment:
Should we handle this in the `PinotFS`?

Reply:
Per https://github.com/apache/incubator-pinot/pull/5890 seems that this is expected behavior.